src/types/ErrorRate.ts

```diff
@@ -0,0 +1,171 @@
+import { NA, RequestHealth } from './Health';
+import { RateHealthConfig, ToleranceConfig } from './ServerConfig';
+import { serverConfig } from '../config/ServerConfig';
+import { ThresholdStatus, getRequestErrorsStatus, RATIO_NA } from './Health';
+
+export interface ErrorRatio {
+  errorRatio: { status: ThresholdStatus; protocol?: string };
+  inboundErrorRatio: { status: ThresholdStatus; protocol?: string };
+  outboundErrorRatio: { status: ThresholdStatus; protocol?: string };
+}
+
+export interface Rate {
+  requestRate: number;
+  errorRate: number;
+  errorRatio: number;
+}
+
+export const regexProtocol = {
+  grpc: new RegExp(/^[1-9]$|^1[0-6]$/),
+  /* eslint-disable-next-line */
+  http: new RegExp(/^[4-5]\d\d$/)
+};
+
+export const checkExpr = (expr: string, value: string): boolean => {
+  if (expr !== '' && value !== '') {
+    const reg = new RegExp(expr);
+    return reg.test(value);
+  }
+  return false;
+};
+
+const getConfig = (ns: string, name: string, kind: string): RateHealthConfig | undefined => {
+  for (let rate of serverConfig.healthConfig.rate) {
```

Comment:
I have something weird happening here. It crashes on my machine, seems like my runtime doesn't like this `let ... for ... of` construct, although it passes at compile time. I'm seeing `TypeError: o is null` in createForOfIteratorHelper.js, which is something from babel runtime.
Has to do with my nodejs version maybe?

Reply:
mm maybe, it's working in my env with npm 6.13.4 and yarn 1.21.1.

It's weird because I have a lot of for let of in aggregate method too :)

Reply:
so it turned out it failed because serverConfig.healthConfig.rate can be null